ngx_http_vod_module.c

```diff
@@ -3512,6 +3516,11 @@ ngx_http_vod_http_reader_open_file(ngx_http_request_t* r, ngx_str_t* path, uint3
 	// Note: for http, no need to open any files, just save the remote uri
 	state->r = r;
 	state->cur_remote_suburi = *path;
+	state->upstream_location = ctx->submodule_context.conf->upstream_location;
+	if (ctx->state == STATE_MAP_OPEN)
```

Comment:
Also, after reversing the if, you'll need to make sure remote_upstream_location is set, if it's not, use the default upstream_location. So it should be something like - if (state == MAP_OPEN || remote_upstream_location.len == 0) use upstream_location, else use remote_upstream_location